### PR TITLE
Fix server list being dropped when invalid ip is provided by a single server

### LINKED ISSUE #161: Domains in the IP field of a hub listing cause the app to silently drop the entire server list.
If *any* of the entries in the serverlist endpoint contain data in the IP field that does *not* resolve to an IP address (either v4 or v6, both work), Stationhub will report no error yet display no servers in the server list

### PR DESCRIPTION
**Pull request by CorruptComputer**
This change should also allow you to join a server using a domain name instead of an IP address. Tested using `natsuko.ninja` for Wild Frontier FFA, and I was able to join and play without issues.

Fixes #161

### CODE REVIEW

File: UnitystationLauncher/Services/ServerService.cs [hunk 1]
```diff
@@ -40,14 +39,14 @@ private async Task<IReadOnlyList<Server>> GetServerListAsync()
...
+            string data = await _http.GetStringAsync(ApiUrls.ServerListUrl);
+            List<Server>? serverData = JsonConvert.DeserializeObject<ServerList>(data)?.Servers;
             Log.Information("Server list fetched");
 
-            var servers = new List<Server>();
+            List<Server> servers = new();
             if (serverData != null)
             {
-                foreach (var server in serverData)
+                foreach (Server server in serverData)
```
Comment (corp-0) [inline UnitystationLauncher/Services/ServerService.cs]: can you validate here that the server has a valid ip/domain and skip them in the listing if not?

Comment (CorruptComputer) [reply UnitystationLauncher/Services/ServerService.cs]: Yeah I can do that later today.

Comment (CorruptComputer) [reply UnitystationLauncher/Services/ServerService.cs]: ServerIp should be validated for a valid IP or a valid domain name now. If it is neither it should get dropped and log a warning.

File: UnitystationLauncher/Models/Api/Server.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.IO;
+using System.Net;
```
Comment (CorruptComputer) [inline UnitystationLauncher/Models/Api/Server.cs:3]: This Codacy issue is a false positive, removing this causes the compile to fail. It is being used on line 63 below.

**Review by corp-0 — APPROVED** [review]

**Pull request merged**